tests/Unit/MerchantCenter/AccountServiceTest.php

```diff
@@ -770,4 +768,101 @@ public function test_disconnect() {
 
 		$this->account->disconnect();
 	}
+
+	public function test_update_wpcom_api_authorization() {
+		$status = 'approved';
+		$nonce  = 'nonce-123';
+
+		$this->options->expects( $this->once() )
+			->method( 'get' )
+			->willReturn( 'nonce-123' );
+
+		$this->options->expects( $this->once() )
+			->method( 'update' )
+			->with( OptionsInterface::WPCOM_REST_API_STATUS, 'approved' );
+
+		$this->options->expects( $this->never() )
+			->method( 'delete' );
+
+		$this->account->update_wpcom_api_authorization( $status, $nonce );
+	}
+
+	public function test_update_wpcom_api_authorization_nonce_not_provided() {
+		$status = 'approved';
+		$nonce  = '';
+
+		$this->options->expects( $this->once() )
+			->method( 'get' )
+			->willReturn( 'nonce-123' );
+
+		$this->options->expects( $this->never() )
+			->method( 'update' );
+
+		$this->options->expects( $this->never() )
+			->method( 'delete' );
+
+		try {
+			$this->account->update_wpcom_api_authorization( $status, $nonce );
```

Comment:
If `$this->account->update_wpcom_api_authorization` pass, then no assertion will happen and the test will pass. 

I saw in the code the we use

`$this->expectException` and `$this->expectExceptionMessage` for avoiding try catch